quixstreams/schema_registry/configuration.py

```diff
@@ -0,0 +1,49 @@
+from typing import Callable, Optional
+
+from pydantic import SecretStr
+from confluent_kafka.schema_registry import topic_subject_name_strategy
+
+from quixstreams.utils.settings import BaseSettings
+from quixstreams.models.serializers import SerializationContext
+
+__all__ = [
+    "SchemaRegistryClientConfig",
+    "SchemaRegistrySerializationConfig",
+    "SchemaRegistryConfig",
+]
+
+SubjectNameStrategy = Callable[[SerializationContext, str], str]
+
+
+class SchemaRegistryClientConfig(BaseSettings):
+    """
+    Configuration required to establish the connection with a Schema Registry.
+    """
+
+    url: str
+    ssl_ca_location: Optional[str] = None
+    ssl_key_location: Optional[str] = None
+    ssl_certificate_location: Optional[str] = None
+    basic_auth_user_info: Optional[SecretStr] = None
+
+
+class SchemaRegistrySerializationConfig(BaseSettings):
+    """
+    Configuration that instructs Serializer how to handle communication with a Schema Registry.
+    """
+
+    auto_register_schemas: bool = True
+    normalize_schemas: bool = False
+    use_latest_version: bool = False
+    subject_name_strategy: SubjectNameStrategy = topic_subject_name_strategy
+
+
+class SchemaRegistryConfig(BaseSettings):
+    """
+    Provides an interface to integrate with a Schema Registry.
+    """
+
+    client: SchemaRegistryClientConfig
+    serialization: SchemaRegistrySerializationConfig = (
+        SchemaRegistrySerializationConfig()
+    )
```

Comment:
Is it necessary to nest the settings this way? It makes the initialization more complicated and requires more imports, etc.
The attributes in these objects don't intersect, so why don't we have a single object with documented initialization params?

Reply:
Also, let's add docstrings for individual params too, so it's clear from the code what needs to be passed

Reply:
I did that for two reasons:

1. Even if they don't intersect, client and serializer will complain if it receives a bigger dict:  [here](https://github.com/confluentinc/confluent-kafka-python/blob/master/src/confluent_kafka/schema_registry/schema_registry_client.py#L118) and [here](https://github.com/confluentinc/confluent-kafka-python/blob/master/src/confluent_kafka/schema_registry/avro.py#L228). This can be mitigated with a bit more verbose `SchemaRegistryConfig.as_dict` call with specific fields to include.
2. Hypothetically someone could have a single `SchemaRegistryClientConfig` config but several distinct `SchemaRegistrySerializationConfig` instances for to pass down to multiple distinct serializers. In this case some attributes in flat SchemaRegistryConfig would have to be duplicated. But all in all, maybe that problem is both, not very probable and not very severe.

Given those considerations, let me know what your preference would be?

Reply:
What if we simply have only two objects then: one for schema registry urls, and another for schema registry params. 
And Serializers will accept two params instead of one.

In this case, you at least don't need a third class two wrap another two.